src/ConnectedMode/Migration/MSBuildFileCleaner.ElementAndAttributeTailMatcher.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    internal partial class MSBuildFileCleaner
+    {
+        private static class ElementAndAttributeTailMatcher
```

Comment:
The AdditionalFile references look like this:
`<AdditionalFiles Include="xxx[partial value]" />`

... and the ruleset Include reference looks like this:
`<Include Path="xxx[partial value]" />`

Logically cleaning up the AdditionalFiles and ruleset Includes follow the same pattern:
- matching first an element name, then an attribute name, then a value in the attribute; so I factored it out into a helper class.